kernels/volk/volk_32f_stddev_and_mean_32f_x2.h

```diff
@@ -73,325 +73,513 @@
-    __m128 squareAccumulator = _mm_setzero_ps();
-    __m128 aVal = _mm_setzero_ps();
-    for(;number < quarterPoints; number++) {
-      aVal = _mm_load_ps(aPtr);                     // aVal = x
-      accumulator = _mm_add_ps(accumulator, aVal);  // accumulator += x
-      aVal = _mm_mul_ps(aVal, aVal);                // squareAccumulator += x^2
-      squareAccumulator = _mm_add_ps(squareAccumulator, aVal);
-      aPtr += 4;
-    }
-    _mm_store_ps(meanBuffer,accumulator); // Store the results back into the C container
-    _mm_store_ps(squareBuffer,squareAccumulator); // Store the results back into the C container
-    newMean = meanBuffer[0];
-    newMean += meanBuffer[1];
-    newMean += meanBuffer[2];
-    newMean += meanBuffer[3];
-    returnValue = squareBuffer[0];
-    returnValue += squareBuffer[1];
-    returnValue += squareBuffer[2];
-    returnValue += squareBuffer[3];
-
-    number = quarterPoints * 4;
-    for(;number < num_points; number++){
-      returnValue += (*aPtr) * (*aPtr);
-      newMean += *aPtr++;
-    }
-    newMean /= num_points;
-    returnValue /= num_points;
-    returnValue -= (newMean * newMean);
-    returnValue = sqrtf(returnValue);
+  if (num_points == 0) { return; }
+
+  const float* in_ptr = inputBuffer;
+
+  unsigned int number = 1;
+
+  __VOLK_ATTR_ALIGNED(16) float T[8] = {0.f};
+  __VOLK_ATTR_ALIGNED(16) float S[8] = {0.f};
+
+  if (num_points < 8) {   
+    T[0] = (*in_ptr++);
+    goto FINALIZE; }
+
+  const unsigned int eigth_points = num_points / 8;
+
+  __m128 T0_acc = _mm_loadu_ps(in_ptr);
+  in_ptr += 4;
+  __m128 T1_acc = _mm_loadu_ps(in_ptr);
+  in_ptr += 4;
+  __m128 S0_acc = _mm_setzero_ps();
+  __m128 S1_acc = _mm_setzero_ps();
+  __m128 v0_reg, v1_reg;
+  __m128 x0_reg, x1_reg;
+  __m128 f_reg;
+
+
+  for(;number < eigth_points; number++) {
+    v0_reg = _mm_loadu_ps(in_ptr);
+    in_ptr += 4;
+    __VOLK_PREFETCH(in_ptr + 4);
+
+    v1_reg = _mm_loadu_ps(in_ptr);
+    in_ptr += 4;    
+    __VOLK_PREFETCH(in_ptr + 4);
+
+    float n   = (float) number;
+    float np1 = n + 1.f;
+    f_reg = _mm_set_ps1(  1.f/( n*np1 ) );
+    
+    T0_acc = _mm_add_ps(T0_acc, v0_reg);
+
+    x0_reg = _mm_set_ps1(np1);
+    x0_reg = _mm_mul_ps(x0_reg, v0_reg);
+    x0_reg = _mm_sub_ps(x0_reg, T0_acc);
+    x0_reg = _mm_mul_ps(x0_reg, x0_reg);
+    x0_reg = _mm_mul_ps(x0_reg, f_reg);
+    S0_acc = _mm_add_ps(S0_acc, x0_reg);
+
+    T1_acc = _mm_add_ps(T1_acc, v1_reg);
+
+    x1_reg = _mm_set_ps1(np1);
+    x1_reg = _mm_mul_ps(x1_reg, v1_reg);
+    x1_reg = _mm_sub_ps(x1_reg, T1_acc);
+    x1_reg = _mm_mul_ps(x1_reg, x1_reg);
+    x1_reg = _mm_mul_ps(x1_reg, f_reg);
+    S1_acc = _mm_add_ps(S1_acc, x1_reg);
+  }
+
+  _mm_store_ps(&T[0], T0_acc);
+  _mm_store_ps(&T[4], T1_acc);
+  _mm_store_ps(&S[0], S0_acc);
+  _mm_store_ps(&S[4], S1_acc);
+
+  accrue_result( S, T, 8, eigth_points);
+
+  number = eigth_points*8;
+
+  FINALIZE:
+  for (; number < num_points; number++) {
```

Comment:
I suggest to bind `number` to the scope of this for loop. Also, I suggest to remove `FINALIZE` as the target for the previous `goto` together with `goto`.
Since this is a rather difficult to catch subtlety how `number` is defined, I suggest to find a more verbose control flow that doesn't rely on `goto` here.

Reply:
I'm changing all instances to 
```
  size_t points_done = eigth_points*8;

  for (; points_done < num_points; points_done++) {
    float val = (*in_ptr);
    PSquareSumLoc[0] = update_square_sum_1_val(PSquareSumLoc[0], PSumsLoc[0], points_done, val);
    PSumsLoc[0]     += val;
    in_ptr++;
  }
```